ext/math_test.go

```diff
@@ -0,0 +1,398 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ext
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/cel-go/cel"
+)
+
+func TestMath(t *testing.T) {
+	mathTests := []struct {
+		expr string
+		in   any
+	}{
+		// Tests for math.least
+		{expr: "math.least(-0.5) == -0.5"},
+		{expr: "math.least(-1) == -1"},
+		{expr: "math.least(1u) == 1u"},
+		{expr: "math.least(42.0, -0.5) == -0.5"},
```

Comment:
It looks like you added overloads for mixed numeric types -- could you add a test for those cases (math.least(1, 1.1) etc)

Reply:
Certainly. Good call. There's some question whether I should just expose the underlying functions directly since they're more friendly to use with the type-checker. Thoughts?

Reply:
Do you mean in addition to the packaged library or instead of? I don't see any problem with exposing the extension function impl for flexibility, but I do prefer the consistency of providing the standard extension library.

Reply:
I'll keep the `@min`, `@max` definitions private then. Tests added.